File: 1768-merge-strings-alternately/1768-merge-strings-alternately.java
class Solution {
    public String mergeAlternately(String w1, String w2) {
        String ans="";
        int i=0;
        int j=0;
        int flag=0;
        while(i< w1.length() && j< w2.length()){
            ans= ans+ w1.charAt(i)+ w2.charAt(j);
            i++;j++;
        }
        if(i== w1.length()){
            ans= ans+ w2.substring(j);
        }else if(j== w2.length()){
            ans= ans+ w1.substring(i);
        }
        return ans;
    }
}

File: 1768-merge-strings-alternately/1768-merge-strings-alternately.scala
object Solution {
    def mergeAlternately(word1: String, word2: String): String = {
        (word1, word2)match{
            case(w1, "")=>w1
            case("", w2)=>w2
            case(w1, w2)=>w1.head.toString + w2.head.toString+ mergeAlternately(w1.tail, w2.tail)
        }
    }
}
